backend/dal/Helpers/Extensions/LeaseExtensions.cs

```diff
@@ -27,13 +28,10 @@ private static IQueryable<Entities.PimsLease> GenerateCommonLeaseQuery(this IQue
                 || l.PimsLeaseTenants.Any(tenant => tenant.Organization != null && EF.Functions.Like(tenant.Organization.OrganizationName, $"%{filter.TenantName}%")));
             }
 
-            if (!string.IsNullOrWhiteSpace(filter.PidOrPin))
+            if (!string.IsNullOrWhiteSpace(filter.PinOrPid))
             {
-                var pidOrPinValue = filter.PidOrPin.Replace("-", "").Trim();
-                if (Int32.TryParse(pidOrPinValue, out int pidOrPin))
-                {
-                    query = query.Where(l => l.PimsPropertyLeases.Where(p => p != null && p.Property != null).Select(p => p.Property).Count(p => p.Pid == pidOrPin || p.Pin == pidOrPin) > 0);
-                }
+                var pinOrPidValue = filter.PinOrPid.Replace("-", "").Trim();
+                query = query.Where(l => l.PimsPropertyLeases.Any(pl => pl != null && (pl.Property.Pid.ToString().Contains(pinOrPidValue) || pl.Property.Pin.ToString().Contains(pinOrPidValue))));
```

Comment:
I think we have been using `EF.Functions.Like(...)` for partial matches. See line 27 above.
I would expect something like:

```cs
  ... && EF.Functions.Like(pl.Property.Pid.ToString(), $"%{pinOrPidValue}%") && ...
```

I'm unsure if using `X.ToString().Contains(Y)` is equivalent. Where you able to test partial matches?

Reply:
Partial matches worked fine using Contains since that is translated to the correct SQL by EF:
```
WHERE ((@__pinOrPidValue_0 LIKE N'') OR 
          	(CHARINDEX(@__pinOrPidValue_0, CONVERT(VARCHAR(11), [p].[PID])) > 0)) OR 
          ((@__pinOrPidValue_0 LIKE N'') OR 
          	(CHARINDEX(@__pinOrPidValue_0, CONVERT(VARCHAR(11), [p].[PIN])) > 0))
```

Reply:
Using like produces a similar one with the conversion
```
WHERE (CONVERT(VARCHAR(11), [p].[PID]) LIKE @__Format_1) OR (CONVERT(VARCHAR(11), [p].[PIN]) LIKE @__Format_1)
```